extended.js

```diff
@@ -21,64 +21,51 @@ Object.defineProperty(exports, "__esModule", { value: true });
+        var result = new BTreeEx(undefined, this._compare, this._maxNodeSize);
+        var target = result;
+        target._root = source._root;
+        target._size = source._size;
+        return result;
     };
     BTreeEx.prototype.greedyClone = function (force) {
-        return wrapBaseTree(_super.prototype.greedyClone.call(this, force));
+        var source = this;
+        var result = new BTreeEx(undefined, this._compare, this._maxNodeSize);
+        var target = result;
+        target._root = source._root.greedyClone(force);
+        target._size = source._size;
+        return result;
     };
     BTreeEx.prototype.with = function (key, value, overwrite) {
-        var result = _super.prototype.with.call(this, key, value, overwrite);
-        return result === this
-            ? this
-            : ensureExtendedTree(result);
+        return _super.prototype.with.call(this, key, value, overwrite);
     };
     BTreeEx.prototype.withPairs = function (pairs, overwrite) {
-        var result = _super.prototype.withPairs.call(this, pairs, overwrite);
-        return result === this ? this : ensureExtendedTree(result);
+        return _super.prototype.withPairs.call(this, pairs, overwrite);
     };
     BTreeEx.prototype.withKeys = function (keys, returnThisIfUnchanged) {
-        var result = _super.prototype.withKeys.call(this, keys, returnThisIfUnchanged);
-        return result === this ? this : ensureExtendedTree(result);
+        return _super.prototype.withKeys.call(this, keys, returnThisIfUnchanged);
     };
     BTreeEx.prototype.without = function (key, returnThisIfUnchanged) {
-        var result = _super.prototype.without.call(this, key, returnThisIfUnchanged);
-        return result === this ? this : ensureExtendedTree(result);
+        return _super.prototype.without.call(this, key, returnThisIfUnchanged);
     };
     BTreeEx.prototype.withoutKeys = function (keys, returnThisIfUnchanged) {
-        var result = _super.prototype.withoutKeys.call(this, keys, returnThisIfUnchanged);
-        return result === this ? this : ensureExtendedTree(result);
+        return _super.prototype.withoutKeys.call(this, keys, returnThisIfUnchanged);
     };
     BTreeEx.prototype.withoutRange = function (low, high, includeHigh, returnThisIfUnchanged) {
-        var result = _super.prototype.withoutRange.call(this, low, high, includeHigh, returnThisIfUnchanged);
-        return result === this ? this : ensureExtendedTree(result);
+        return _super.prototype.withoutRange.call(this, low, high, includeHigh, returnThisIfUnchanged);
     };
     BTreeEx.prototype.filter = function (callback, returnThisIfUnchanged) {
-        var result = _super.prototype.filter.call(this, callback, returnThisIfUnchanged);
-        return result === this ? this : ensureExtendedTree(result);
+        return _super.prototype.filter.call(this, callback, returnThisIfUnchanged);
     };
     BTreeEx.prototype.mapValues = function (callback) {
```

Comment:
These functions shouldn't exist at all.

Reply:
Check out my latest commit, I changed the methods that I could (those that didn't change key/value types) to return this (it required casts in the base class, but that is safe) and I removed those specific methods from the BTreeEx class. Is that what you were hoping for?